solidity/contracts/ECDSARewards.sol

```diff
@@ -266,6 +294,32 @@ contract ECDSARewards is Rewards {
         }
     }
 
+    function calculateLifespanReward(bytes32 _keep, uint256 amount)
+        internal returns (uint256) {
+        uint256 createdAt = _getCreationTime(_keep);
+
+        Heartbeat lastHeartbeat = keepHeartbeats[toAddress(_keep)];
+
+        uint256 lastTimestamp = uint256(lastHeartbeat.timestamp);
+        uint256 keepBond = uint256(lastHeartbeat.bondAmount);
+
+        uint256 rewardLifespan = lastTimestamp == 0
+            ? assumedMinimumLifetime
+            : lastTimestamp.sub(createdAt);
+
+        uint256 rewardBond = keepBond == 0
+            ? assumedMinimumBond
```

Comment:
OK, and here I see the first heartbeat is needed because otherwise the minimum bond can be assumed. Still, I think both `assumedMiniumumLifetime` and `assumedMinimumBond` make sense.